CONTRIBUTING.adoc

```diff
@@ -0,0 +1,184 @@
+
+* `/assign @user1 @user2` to assign an issue or PR to specific people for review or approval.
+* `/lgtm` and `/approve` to approve a PR. Note that _anyone_ may `/lgtm` a PR, but only someone listed in an `OWNERS` file may `/approve` the PR. A PR needs both an approval and an LGTM -- the `/lgtm` review is a good opportunity for non-approvers to practice and develop reviewing skills. `/lgtm` is removed when a PR is updated, but `/approve` is sticky -- once applied, anyone can supply an `/lgtm`.
+* Both Prow (legacy) and GitHub actions (preferred) can run tests on PRs; once all tests are passing and a PR has the `lgtm` and `approved` labels, Prow will submit the PR automatically.
+* You can also use Prow to manage labels on PRs with `/kind ...`, `/good-first-issue`, or `/area ...`
+* See <<branches>> for details on the `/cherrypick` command.
+
+[NOTE]
+====
+As an experimental process, tag tech writer rewritten documents with `reviewed: $DATE` (fill in the date of starting the PR). This allows finding documents which have not been reviewed with the `find` command (Mac and Linux):
+[source,shell]
+----
+find docs -type f -exec /bin/bash -c 'head -1 {} | grep ^--- >/dev/null && grep -L reviewed: {}' \;
+----
+(This currently finds all documents, but as docs are reviewed, will become more useful.)
+In the future, we may also be able to use this metadata to provide freshness information on the website.
+
+====
+
+== Engineer Guide
+
+=== Putting your docs in the right place
+
+There are currently two general types of Knative docs, either contributor related content, or external-facing user content.
+
+==== Contributor-focused content
+
+* _Documentation_: Includes content that is component specific and relevant only to contributors of a given component. Contributor focused documentation is located in the corresponding `docs` folder of that component’s repository. For example, if you contribute code to the Knative Serving component, you might need to add contributor focused information into the `docs` folder of the https://github.com/knative/serving/tree/master/docs/[knative/serving repo].
+
+* _Code samples_: Includes contributor related code or samples. Code or samples that are contributor focused also belong in their corresponding component’s repo. For example, Eventing specific test code is located in the https://github.com/knative/eventing/tree/master/test[knative/eventing tests] folder.
+
+==== User-focused content
```

Comment:
Hey shouldn't we mention docs that are generated from code comments? Remember the kerfuffle with the serving docs I was trying to improve?

Reply:
Adding that to the tech writer guide under "Documentation Structure".